lib/ansible/template/__init__.py

```diff
@@ -482,6 +499,16 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
             else:
                 return variable
 
+    def is_template(self, data):
+        ''' lets us know if data has a template'''
+        try:
+            new = self.template(data)
+        except UndefinedError:
+            return True
```

Comment:
{% what %} is not a valid template but {{foo}} is, it will give a syntax error on templating, not sure on which side this should fall:
a) it has 'tempating markers'
b) its not a valid jinja2 template